### ISSUE TITLE
VarFieldChecker false positive on var local

### ISSUE THREAD

**Issue by sokrahta** [scalastyle/scalastyle#132]
VarFieldChecker reported this var local **pendingIterator** as a var field. We'd like to prohibit var fields but allow var locals, as is suggested by the two checker names.

```
  def arrayOfResults(in: BufferedIterator[JsonEvent], alreadyInArray: Boolean = false): Iterator[ReportItem] = {
    if(!alreadyInArray) expectStartOfArray(in)
    new Iterator[ReportItem] {
      var pendingIterator: Iterator[JValue] = null
      def invalidatePending() {
        if((pendingIterator ne null) && pendingIterator.hasNext) {
          throw new Exception("Row result was not completely consumed")
        }
        pendingIterator = null
      }
      def hasNext = { invalidatePending(); in.hasNext && !in.head.isInstanceOf[EndOfArrayEvent] }
      def next(): ReportItem = {
        if(!hasNext) Iterator.empty.next()
        if(in.head.isInstanceOf[StartOfArrayEvent]) {
          pendingIterator = JsonArrayIterator[JValue](in)
          UpsertReportItem(pendingIterator)
        } else {
          skipNextDatum(in)
          OtherReportItem
        }
      }
    }
  }
```

**Comment by matthewfarwell** [comment]
In the above, pendingIterator is a field, it isn't a var local.
